### PR TITLE
Style/10 gráfica votación

### LINKED ISSUE #10: Prototipo de gráficas en tiempo real del panel de control
Se busca implementar un prototipo inicial de las gráficas en tiempo real que se usarán para su posterior implementación completa, para mostrar información relevante del sistema, ya que actualmente la aplicación no dispone de elementos gráficos que permitan ver las estadísticas de manera rápida y cómoda.

La solución propuesta ha sido contener en una vista utilizando la tecnología de React las distintas gráficas , las cuales harán uso de las votaciones y los votos existentes en la aplicación para mostrar calculos estadísticos de los mismos. 

La alternativa propuesta ha sido realizar una vista haciendo uso de la tecnología de React que mediante un campo de entrada se seleccione la votación y tras esto, se muestren todas las gráficas relacionadas  con la información en tiempo real.

### PR DESCRIPTION
**Pull request by NoSoyDani**

### CODE REVIEW

File: decide_panel/src/components/Bar.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+                label: "Votos",
+                data: [1,1],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
+                    'rgba(54, 162, 235, 0.2)'
+                    
+                ],
+                borderColor: [
+                    'rgba(255, 99, 132, 1)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:22]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(255, 99, 132, 1)",
```

File: decide_panel/src/components/Bar.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  return <div>
+      <Bar
+        data={{
+            labels: ["Opcion 1", "Opcion 2"],
+            datasets: [{
+
+                label: "Votos",
+                data: [1,1],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:17]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(255, 99, 132, 0.2)",
```

File: decide_panel/src/components/Bar.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import Api from "../services/backend";
+import React from "react";
+import { Bar } from "react-chartjs-2";
+import Chart from 'chart.js/auto';
```
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
import Chart from "chart.js/auto";
```

File: decide_panel/src/components/Bar.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                label: "Votos",
+                data: [1,1],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
+                    'rgba(54, 162, 235, 0.2)'
+                    
+                ],
+                borderColor: [
+                    'rgba(255, 99, 132, 1)',
+                    'rgba(54, 162, 235, 1)'
```
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:23]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(54, 162, 235, 1)"
```

File: decide_panel/src/components/Bar.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      <Bar
+        data={{
+            labels: ["Opcion 1", "Opcion 2"],
+            datasets: [{
+
+                label: "Votos",
+                data: [1,1],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
+                    'rgba(54, 162, 235, 0.2)'
```
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:18]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(54, 162, 235, 0.2)"
```

File: decide_panel/src/components/Bar.js (hunk 3 above)
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
import Chart from "chart.js/auto";
```

File: decide_panel/src/components/Graphics.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    <div>
+        <Pie
+          data={{
+              labels: ["Votos"],
+              datasets: [{
+  
+                  label: "",
+                  data: tally,
+                  backgroundColor: [
+                      'rgba(255, 99, 132, 0.2)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(255, 99, 132, 0.2)",
```

File: decide_panel/src/components/Bar.js (hunk 1 above)
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:22]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(255, 99, 132, 1)",
```

File: decide_panel/src/components/Bar.js (hunk 4 above)
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:23]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(54, 162, 235, 1)"
```

File: decide_panel/src/components/Graphics.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                      'rgba(250, 99, 250, 0.2)',
+                      'rgba(250, 0, 0, 0.2)'
+                      
+                  ],
+                  borderColor: [
+                      'rgba(255, 99, 132, 1)',
+                      'rgba(255, 99, 132, 1)',
+                      'rgba(255, 99, 132, 1)',
+                      'rgba(255, 99, 132, 1)',
+                      'rgba(255, 99, 132, 1)'
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:93]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(255, 99, 132, 1)"
```

File: decide_panel/src/components/Bar.js (hunk 2 above)
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:17]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(255, 99, 132, 0.2)",
```

File: decide_panel/src/components/Graphics.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+              datasets: [{
+  
+                  label: "",
+                  data: tally,
+                  backgroundColor: [
+                      'rgba(255, 99, 132, 0.2)',
+                      'rgba(50, 250, 250, 0.2)',
+                      'rgba(23, 99, 250, 0.2)',
+                      'rgba(250, 99, 250, 0.2)',
+                      'rgba(250, 0, 0, 0.2)'
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:85]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(250, 0, 0, 0.2)"
```

File: decide_panel/src/components/Graphics.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from "react";
+import Api from "../services/backend";
+import { ProgressBar } from "primereact/progressbar";
+import React from "react";
+import { Pie } from "react-chartjs-2";
+import Chart from 'chart.js/auto';
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
import Chart from "chart.js/auto";
```

File: decide_panel/src/components/Graphics.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+              labels: ["Votos"],
+              datasets: [{
+  
+                  label: "",
+                  data: tally,
+                  backgroundColor: [
+                      'rgba(255, 99, 132, 0.2)',
+                      'rgba(50, 250, 250, 0.2)',
+                      'rgba(23, 99, 250, 0.2)',
+                      'rgba(250, 99, 250, 0.2)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:84]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(250, 99, 250, 0.2)",
```

File: decide_panel/src/components/Graphics.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        <Pie
+          data={{
+              labels: ["Votos"],
+              datasets: [{
+  
+                  label: "",
+                  data: tally,
+                  backgroundColor: [
+                      'rgba(255, 99, 132, 0.2)',
+                      'rgba(50, 250, 250, 0.2)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:82]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(50, 250, 250, 0.2)",
```

File: decide_panel/src/components/Graphics.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+                      'rgba(23, 99, 250, 0.2)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:83]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(23, 99, 250, 0.2)",
```

File: decide_panel/src/components/Graphics.js [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+                      'rgba(250, 99, 250, 0.2)',
+                      'rgba(250, 0, 0, 0.2)'
+                      
+                  ],
+                  borderColor: [
+                      'rgba(255, 99, 132, 1)',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js:89]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                      "rgba(255, 99, 132, 1)",
```

File: decide_panel/src/components/Graphics.js [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+import { ProgressBar } from "primereact/progressbar";
+import React from "react";
+import { Pie } from "react-chartjs-2";
+import Chart from 'chart.js/auto';
+
+
+const Graphics = () => {
+  const [errorConection, setErrorConection] = useState(null);
+  const [state, setState] = useState({
+    data: '{}',
```
Comment (eveyugyug) [inline decide_panel/src/components/Graphics.js]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
    data: "{}",
```

File: decide_panel/src/components/Bar.js (hunk 5 above)
Comment (eveyugyug) [inline decide_panel/src/components/Bar.js:18]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Tortuga/decide-full-tortuga-almacenamiento/pullRequest?prid=8645779)
```suggestion
                    "rgba(54, 162, 235, 0.2)"
```

**Review by dancasnar1 — APPROVED** [review]
Se ha comprobado la correcta visualización de la página.

**Review by manmunesp12 — APPROVED** [review]
Se ha comprobado la correcta creación de los gráfico referentes a las diferentes votaciones

**Pull request merged**